what4/src/What4/Expr/Builder.hs

```diff
@@ -19,6 +19,20 @@ threads may still clobber state others have set (e.g., the current
 program location) so the potential for truly multithreaded use is
 somewhat limited.  Consider the @exprBuilderFreshConfig@ or
 @exprBuilderSplitConfig@ operations if this is a concern.
+
+-- * Boolean expressions
+
+'ExprBuilder' tries to rewrite expressions in order to keep them as simple
+and concrete as possible. In particular, here are a few considerations for
+boolean-typed expressions:
+
+* Disjunctions are implicitly represented as negated conjunctions
+* Conjunctions are represented via 'BM.ConjMap' (see docs on that type)
+* @xor@ is represented as the negation of equality
+
+Boolean expressions are expected to be somewhat normalized at all times.
+For example, there should never be a double negation (nested 'NotPred').
+See @isNormal@ in @test/Bool.hs@ for the exact expectations.
```

Comment:
I would argue that we should spell out the expectations for how boolean normalization should work here in the library Haddocks, not tucked away in the comments of a test suite. Do you agree?

Reply:
If you agree with the direction of #282, I might like to defer this to an issue for now. I think that this is already an improvement over the status quo (no documentation, no tests), and the solution to #282 will change the invariants.

Reply:
I'm OK with deferring to an issue, as well as the general direction of #282.

Reply:
:+1: #284